scene/jms_util.py

```diff
@@ -20,3 +20,32 @@ def set_translation_from_jms(scene_object, jms_piece, scale=1.0):
 		x=jms_piece.pos_x,
 		y=jms_piece.pos_y,
 		z=jms_piece.pos_z)
+
+def get_absolute_node_transforms_from_jms(node_list):
+	'''
+	Takes a JmsNodes list and returns the absolute transformations in a dict.
+	'''
+	node_transforms = {}
+
+	for i, node in enumerate(node_list):
+		translation = Vector((node.pos_x, node.pos_y, node.pos_z))
+		rotation = Quaternion((-node.rot_w, node.rot_i, node.rot_j, node.rot_k))
+
+		if node.parent_index >= 0:
+			parent = node_transforms[node.parent_index]
+
+			abs_rotation    = parent['rotation'] @ rotation
+			# Rotating this way has the result end up in the instance
+			# we're calling this method from.
+			translation.rotate(parent['rotation'])
+			abs_translation = parent['translation'] + translation
+		else:
+			abs_translation = translation
+			abs_rotation    = rotation
+
+		node_transforms[i] = {
```

Comment:
You define this as a dict, but you're using it as an array. Is there any reason for that?

Reply:
This is really just to keep it in line with the implementation in the nodes function. The nodes function does this for the potential case that there is missing indices. It also allows us to leverage dictionary functions like get and setdefault to avoid manual bounds checks.